src/main/java/seedu/address/logic/commands/UntagCommand.java

```diff
@@ -23,53 +24,70 @@ public class UntagCommand extends Command {
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Untags a tag from the person identified by the index number used in the displayed person list. \n "
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "[" + PREFIX_TAG + "TAG]\n"
-            + "Example: " + COMMAND_WORD + " 1 "
-            + PREFIX_TAG + "bride's side";
+            + ": Untags the guest(s) identified by the index number(s) used in the displayed guest list "
+                + "with the predefined tag(s). \n"
+            + "Parameters: INDEX... (must be positive integer(s))\n"
+            + "[" + PREFIX_TAG + "TAG]... (must be a tag on the guest)\n"
+            + "Example: " + COMMAND_WORD + " 1 2 "
+            + PREFIX_TAG + "bride's side" + " "
+            + PREFIX_TAG + "groom's side";
 
-    public static final String MESSAGE_UNTAG_PERSON_SUCCESS = "Untagged Person: %1$s";
-    public static final String MESSAGE_TAG_NOT_FOUND = "This person does not have this tag.";
+    public static final String MESSAGE_UNTAG_PERSON_SUCCESS = "Untagged guest(s):\n";
+    public static final String MESSAGE_TAG_NOT_FOUND = "Some guest(s) do not have the tag(s): ";
 
-    private final Index targetIndex;
-    private final Tag tag;
+    private final List<Index> targetIndexes;
+    private final Set<Tag> tags;
 
     /**
-     * @param targetIndex of the person in the filtered person list to untag
-     * @param tag to remove from the person
+     * @param targetIndexes of the guest in the filtered person list to untag
+     * @param tags set of tags to remove from the guest
      */
-    public UntagCommand(Index targetIndex, Tag tag) {
-        requireNonNull(targetIndex);
-        requireNonNull(tag);
+    public UntagCommand(List<Index> targetIndexes, Set<Tag> tags) {
+        requireNonNull(targetIndexes);
+        requireNonNull(tags);
 
-        this.targetIndex = targetIndex;
-        this.tag = tag;
+        this.targetIndexes = targetIndexes;
+        this.tags = tags;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        Set<Tag> missingTags = new HashSet<>();
+        StringBuilder successMessage = new StringBuilder();
+        StringBuilder finalMessage = new StringBuilder();
+
+        for (Index targetIndex : targetIndexes) {
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+            Person personToUntag = lastShownList.get(targetIndex.getZeroBased());
+            Set<Tag> newTags = new HashSet<>(personToUntag.getTags());
+            boolean isUpdated = false;
```

Comment:
What if the 1st tag is correctly added but the subsequent tags are not? Would the isUpdated variable still be true? Should it be changed to `isUpdated &= ...` so the user will be notified if 1 or more tags were not correctly updated?

Reply:
This part of the code is within the loop that checks for each index i.e. each guest. For example, tag 1 2 t/tag1 t/tag2, if tag1 is successfully untagged from both guests at indexes 1 and 2 while tag2 is only successfully untagged from index 2, then the message displayed to the user will indicate that some guest(s) does not have the tag 'tag2 '